manifests/plugin/tasks.pp

```diff
@@ -7,19 +7,17 @@
 # $package:: Package name to install, use ruby193-rubygem-foreman-tasks on Foreman 1.8/1.9 on EL
 #            type:String
 #
-# $service:: Service name
-#            type:String
+# $dynflow_in_core:: Whether Foreman ships the Dynflow executor service (true on 1.15+)
+#                    type:Boolean
 #
 class foreman::plugin::tasks(
   $package = $foreman::plugin::tasks::params::package,
-  $service = $foreman::plugin::tasks::params::service,
+  $dynflow_in_core = $::foreman::dynflow_in_core,
 ) inherits foreman::plugin::tasks::params {
   foreman::plugin { 'tasks':
     package => $package,
-  } ~>
-  service { 'foreman-tasks':
-    ensure => running,
-    enable => true,
-    name   => $service,
+  }
+  unless $dynflow_in_core {
```

Comment:
I'm not sure this parameter is actually needed. Given there's an include in ```foreman::service``` if it's true that means it's going to be included anyway.

You're also losing the notify on the service if the plugin changes.